src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -550,10 +569,6 @@ public static CQLStatement getStatement(String queryStr, ClientState clientState
         Tracing.trace("Parsing {}", queryStr);
         CQLStatement.Raw statement = parseStatement(queryStr);
 
-        // Set keyspace for statement that require login
```

Comment:
because this happens in prepare method anyway. It is redundant and in fact I suppose `BatchStatement.Parsed` extends `QualifiedStatement` just because of this call